From 2873f6b05562e167b5c15714d2ce628ff1b5e8af Mon Sep 17 00:00:00 2001
From: Abhishek Singh Chouhan <abhishekchouhan121@gmail.com>
Date: Thu, 23 Feb 2017 18:13:06 +0530
Subject: [PATCH 346/362] HBASE-17682 Region stuck in merging_new state
 indefinitely

Signed-off-by: Andrew Purtell <apurtell@apache.org>
(cherry picked from commit 4caed356f15fec6ace0f5e7641e7526d1e01f7bb)

Change-Id: Ic2cfe326b60018dd189ceea7e1c5be07c538f4f6
Reason:    Supportability
Author:    Abhishek Singh Chouhan <abhishekchouhan121@gmail.com>
Ref:       CDH-53564
---
 .../apache/hadoop/hbase/master/RegionStates.java   |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionStates.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionStates.java
index efb9f48..bfdfecc 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionStates.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/RegionStates.java
@@ -690,7 +690,7 @@ public class RegionStates {
             LOG.info("Found region in " + state +
               " to be reassigned by ServerCrashProcedure for " + sn);
             rits.add(hri);
-          } else if(state.isSplittingNew()) {
+          } else if(state.isSplittingNew() || state.isMergingNew()) {
             regionsToCleanIfNoMetaEntry.add(state.getRegion());
           } else {
             LOG.warn("THIS SHOULD NOT HAPPEN: unexpected " + state);
-- 
1.7.9.5

